File: go/gin/internal/modules/manager/repository.go
package manager

import (
	"context"
	"tcc/internal/database"
	"tcc/internal/modules/manager/entity"
	"time"
)

type ManagerRepository struct {
}

func (r *ManagerRepository) FindByEmail(email string) (*entity.ManagerEntity, error) {
	query := `SELECT * FROM "Gerente" WHERE email = $1 LIMIT 1`
	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
	defer cancel()
	row := database.Conn.QueryRow(ctx, query, email)

	var manager entity.ManagerEntity
	var dataDeNascimento time.Time
	err := row.Scan(
		&manager.ID,
		&manager.IDAgencia,
		&manager.Nome,
		&manager.CPF,
		&manager.Telefone,
		 &dataDeNascimento,
		&manager.Email,
		&manager.Senha,
		&manager.DataDeCriacao,
		&manager.DataDeAtualizacao,
	)

	if err != nil {
		return nil, err
	}

	manager.DataDeNascimento = dataDeNascimento.Format(time.DateOnly)

	return &manager, nil
}


func (r *ManagerRepository) FindAll(quantidade int) ([]*entity.ManagerEntity, error) {
	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
	defer cancel()
	rows, err := database.Conn.Query(ctx,
		`SELECT * FROM "Gerente" LIMIT $1`,
		quantidade,
	)
	if err != nil {
		return nil, err
	}
	defer rows.Close()

	var managers []*entity.ManagerEntity
	for rows.Next() {
		var manager entity.ManagerEntity
    var dataDeNascimento time.Time
		err := rows.Scan(
			&manager.ID,
			&manager.IDAgencia,
			&manager.Nome,
			&manager.CPF,
			&manager.Telefone,
			&dataDeNascimento,
			&manager.Email,
			&manager.Senha,
			&manager.DataDeCriacao,
			&manager.DataDeAtualizacao,
		)
		if err != nil {
			return nil, err
		}

		manager.DataDeNascimento = dataDeNascimento.Format(time.DateOnly)

		managers = append(managers, &manager)
	}
	return managers, nil
}

File: python/flask/src/app/modules/managers/repository.py
from typing import Optional
from sqlalchemy.sql import text

# Common
from app.database import db

# Managers
from .entity import ManagerEntity

class ManagersRepository:
    def find_by_email(self, email: str) -> Optional[ManagerEntity]:
        with db.connect() as connection:
            result = connection.execute(text("""
                SELECT 
                    m.id, 
                    m."idAgencia" AS "idAgencia",
                    m.nome AS nome,
                    m.cpf AS cpf,
                    m.telefone AS telefone,
                    m."dataDeNascimento" AS "dataDeNascimento",
                    m.email AS email,
                    m.senha AS senha,
                    m."dataDeCriacao" AS "dataDeCriacao",
                    m."dataDeAtualizacao" AS "dataDeAtualizacao"
                FROM "Gerente" m
                WHERE m.email = :email LIMIT 1
            """), {"email": email})
            row = result.mappings().first()
            return dict(row) if row else None

    def find_all(self, quantidade: int):
        with db.connect() as connection:
            result = connection.execute(text("""
                SELECT * FROM "Gerente" LIMIT :quantidade
            """), {"quantidade": quantidade})
            rows = result.mappings().all()
            return [dict(row) for row in rows]